plugins/catalog-backend/src/search/defaultCatalogCollatorEntityTransformer.test.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { defaultCatalogCollatorEntityTransformer } from './defaultCatalogCollatorEntityTransformer';
+
+const entity = {
+  apiVersion: 'backstage.io/v1alpha1',
+  kind: 'Component',
+  metadata: {
+    title: 'Test Entity',
+    name: 'test-entity',
+    description: 'The expected description',
+    namespace: 'namespace',
+  },
+  spec: {
+    type: 'some-type',
+    lifecycle: 'experimental',
+    owner: 'someone',
+  },
+};
+
+const userEntity = {
+  apiVersion: 'backstage.io/v1alpha1',
+  kind: 'User',
+  metadata: {
+    name: 'test-user-entity',
+    description: 'The expected user description',
+  },
+  spec: {
+    profile: {
+      displayName: 'User 1',
+    },
+  },
+};
+
+describe('DefaultCatalogCollatorEntityTransformer', () => {
+  describe('transform', () => {
+    it('maps a returned entity', async () => {
+      const document = defaultCatalogCollatorEntityTransformer(entity);
+
+      expect(document).toMatchObject({
+        title: entity.metadata.title,
+        text: entity.metadata.description,
+        namespace: entity.metadata.namespace,
+        componentType: entity.spec.type,
+        lifecycle: entity.spec.lifecycle,
+        owner: entity.spec.owner,
+      });
+    });
+
+    it('maps a returned entity with default fallback', async () => {
```

Comment:
What happens if I do not call the default transformer 🤔 ? Maybe we can always set the default title and text in case they are not returned by the transformer function, as I commented above?

Reply:
TS will shout on you :D if you don't call default transformer you need to comply with basic document anyway